tasks/move-granules/index.js

```diff
@@ -321,54 +283,45 @@ async function updateEachCmrFileAccessURLs(cmrFiles, granulesObject, distEndpoin
  * @param {Object} event - Lambda function payload
  * @param {Object} event.config - the config object
  * @param {string} event.config.bucket - Bucket name where public/private keys are stored
- * @param {string} event.config.granuleIdExtraction - regex needed to extract granuleId
- *                                                    from filenames
- * @param {Array} event.config.input_granules - an array of granules
+ * @param {Object} event.config.buckets - Buckets config
  * @param {string} event.config.distribution_endpoint - distribution endpoint for the api
  * @param {Object} event.config.collection - collection configuration
  *                     https://nasa.github.io/cumulus/docs/data-cookbooks/setup#collections
  * @param {boolean} [event.config.moveStagedFiles=true] - set to false to skip moving files
  *                                 from staging to final bucket. Mostly useful for testing.
- * @param {Array} event.input - an array of s3 uris
+ * @param {Object} event.input - an array of s3 uris
  * @returns {Promise} returns the promise of an updated event object
  */
 async function moveGranules(event) {
   // we have to post the meta-xml file of all output granules
   // first we check if there is an output file
   const config = get(event, 'config');
-  const sourceBucket = get(config, 'bucket'); // the name of the bucket with staged data.
   const bucketsConfig = new BucketsConfig(get(config, 'buckets'));
-  const granuleIdExtractionRegex = get(config, 'granuleIdExtraction', '(.*)');
-  const inputGranules = get(config, 'input_granules', {});
-  const distEndpoint = get(config, 'distribution_endpoint');
   const moveStagedFiles = get(config, 'moveStagedFiles', true);
-  const collection = config.collection;
 
   const duplicateHandling = duplicateHandlingType(event);
 
-  const inputFileList = get(event, 'input', []);
-
-  // Get list of cmr file objects from the input Array of S3 filenames (in
-  // staging location after processing)
-  const cmrFiles = getCmrFiles(inputFileList, granuleIdExtractionRegex);
-
-  const allGranules = mergeInputFilesWithInputGranules(
-    inputFileList, inputGranules, granuleIdExtractionRegex
-  );
+  const granulesInput = get(event, 'input.granules', []);
+  const cmrFiles = reduceGranulesToCmrFileObjects(granulesInput);
```

Comment:
I'm sort of confused. I thought this PR was updating `move-granules` to expect a granules object as input, whereas this code seems to be transforming the array of files input into a granules object?

Reply:
Or this the opposite: converting the granules object back into an array of files?

Reply:
It reduce the granules object down to an array of CMR files for the downstream functions. I've been looking into updating and simplifying those as well, I've just not yet had the time to dedicate to it, since it's a pretty big overhaul. I'd like to include it in this ticket, though.

Reply:
I should note that the important thing (imho) is that `move-granules` no longer depends on the `granuleIdExtraction` assumption to associate files and granules, and that is something that is already achieved by this PR, but simplifying those downstream functions would be nice because they are very large, and I'm looking at them now.

Reply:
Yeah I don't think that change is necessary in this work per se and certainly not in this PR but I'll leave that to your discretion